crates/hyle-modules/src/modules/prover.rs

```diff
@@ -104,65 +106,93 @@ pub enum AutoProverEvent<Contract> {
     /// Event sent when a blob is executed as success
     /// proof will be generated & sent to the node
     SuccessTx(TxHash, Contract),
+    /// Event sent in order to add an AutoProver for that contract
+    ProveContract(ContractName, Contract),
 }
 
-impl<Contract> Module for AutoProver<Contract>
+impl<Contract> Module for AutoProverModule<Contract>
 where
     Contract: TxExecutorHandler + BorshDeserialize + Debug + Send + Sync + Clone + 'static,
 {
     type Context = Arc<AutoProverCtx<Contract>>;
 
     async fn build(bus: SharedMessageBus, ctx: Self::Context) -> Result<Self> {
-        let bus = AutoProverBusClient::<Contract>::new_from_bus(bus.new_handle()).await;
-
-        let file = ctx
-            .data_directory
-            .join(format!("autoprover_{}.bin", ctx.contract_name).as_str());
-
-        let store = match Self::load_from_disk::<AutoProverStore<Contract>>(file.as_path()) {
```

Comment:
the load_from_disk is important, you should not remove it. It was not well done, #1217 fixes it, but we definitively needs it. You will need to update AutoProver to keep a `store` I think